### PR TITLE
Doc: fix steps of deploying image to oc cluster

### PR DESCRIPTION
**Pull request by huikang**

### CODE REVIEW

**Comment by huikang** [comment]
I tested pushing image without `oc login`, but it does not work and gave me the following error:

```
REMOTE_REGISTRY=true PUSH_USER=kubeadmin SKIP_BUILD=true \
PUSH_PASSWORD=sqCa9-zMFds-c2Rd2-rt7i7 KUBECONFIG=/root/aws-oc-clusters/kubeconfig DEBUG=true \
make deploy-image


Error from server (Forbidden): services "image-registry" is forbidden: User "system:anonymous" cannot get resource "services" in API group "" in the namespace "openshift-image-registry"
```

File: docs/HACKING.md [hunk 1]
```diff
@@ -238,15 +238,11 @@ You'll need `podman` in addition to `docker`.
 If you used the new `openshift-installer`, it creates a user named `kubeadmin`
 with the password in the file `installer/auth/kubeadmin_password`.
 
-To push the image to the remote registry, set `REMOTE_REGISTRY=true`, `PUSH_USER=kubeadmin`,
-`KUBECONFIG=/path/to/installer/auth/kubeconfig`, and
-`PUSH_PASSWORD=$( cat /path/to/installer/auth/kubeadmin_password )`.  You do not have to
-`oc login` to the cluster.  If you already built the image, use `SKIP_BUILD=true` to do
-a push only.
+To push the image to the remote registry, set `REMOTE_REGISTRY=true`. If you already built
```
Comment (jcantrill) [inline docs/HACKING.md:241]: i dont believe REMOTE_REGISTRY is needed anylonger

**Review by jcantrill — COMMENTED** [review]
/approve

File: docs/HACKING.md (hunk 1 above)
Comment (huikang) [reply docs/HACKING.md:241]: Hi, @jcantrill you are right. The variable `REMOTE_REGISTRY` appears in `./hack/common`, which is not sourced by `make deploy-image`.

File: docs/HACKING.md [hunk 2]
```diff
@@ -238,15 +238,10 @@ You'll need `podman` in addition to `docker`.
...
-a push only.
+If you already built the image with `make image`, use `SKIP_BUILD=true` to do a push
+only. You need `oc login` to the cluster before pushing the image to the register.
 
 ```bash
-REMOTE_REGISTRY=true PUSH_USER=kubeadmin SKIP_BUILD=true \
-KUBECONFIG=/path/to/installer/auth/kubeconfig \
-PUSH_PASSWORD=$( cat /path/to/installer/auth/kubeadmin_password ) \
-make deploy-image
+oc login --token=<token> --server=https://api.<cluster>:6443
```
Comment (ewolinetz) [inline docs/HACKING.md:245]: if we are using the openshift installer we can also log in using `oc login -u kubeadmin -p <password>`

**Comment by ewolinetz** [comment]
one quick nit to offer an additional way to log in to the cluster, otherwise :+1:

**Comment by huikang** [comment]
> one quick nit to offer an additional way to log in to the cluster, otherwise 👍

Fixed. Thanks for your comment.

**Pull request merged**